sql/core/src/main/scala/org/apache/spark/sql/DataFrameImpl.scala

```diff
@@ -112,6 +112,12 @@ private[sql] class DataFrameImpl protected[sql](
 
   override def printSchema(): Unit = println(schema.treeString)
 
+  override def printPlan(extended: Boolean): Unit = {
+    ExplainCommand(
+      logicalPlan,
+      extended = extended).rdd.collect().map(r => println(r.getString(0)))
```

Comment:
Yeah, you're right, a little bit tricky thing here is `collect` is the API of `TreeNode`, and `ExplanCommand` is also a `TreeNode`, so it will causes ambiguous when implicitly convert to `DataFrame`, anyway, I will do that explicitly.